src/index.ts

```diff
@@ -90,20 +93,47 @@ export default class ReactTSTarget extends Target {
         });
 
         engine.registerHelper('ifValueType', (type, options) => {
-            if ((type?.type || type?.ref) && type?.type?.toLowerCase() !== 'void' && type?.ref?.toLowerCase() !== 'void') {
+            if (
+                (type?.type || type?.ref) &&
+                type?.type?.toLowerCase() !== 'void' &&
+                type?.ref?.toLowerCase() !== 'void'
+            ) {
                 return Template.SafeString(options.fn());
             }
             return Template.SafeString('');
         });
 
+        engine.registerHelper('ifVoidType', (type: TypeLike, options: HelperOptions) => {
+            // Check if type has a property 'string' (which it would have if it is a SafeString)
+            if (type && typeof type === 'object' && 'string' in type) {
+                type = type.string as string;
+            }
+
+            let isVoid = false;
+            if (typeof type === 'string') {
```

Comment:
there is a "typeName(type)" helper which will do this for you

Reply:
740ae9bb97f26d05a86cdd8058af83160c2e52a2 Here is an updated version where I use that version... but I'm not sure it is much cleaner. The `TypeLike` in codegen-target is not the same type as `TypeLike` in schemas.

```
export type TypeLike = string | {
    type?: string;
    ref?: string;
};
```
vs
```
export interface TypeLike {
    type?: string;
    ref?: string;
}
```